lib/primitives/mtx.js

```diff
@@ -868,6 +868,51 @@ class MTX extends TX {
     return null;
   }
 
+  /**
+   * Sign custom P2WSH input.
+   * @param {Number} index - Index of input being signed.
+   * @param {KeyRing[]} rings - with private key.
+   * Sorted by desired signature order.
+   * @param {Script} redeemScript - its hash is in spending output.
+   * @param {Buffer|Number[]} witnessScript - signatures template.
+   * witnessScript is the SegWit equivalent of scriptSig.
+   * Put an empty buffer as placeholder for each signature.
+   * @param {SighashType} type
+   * @param {Coin|Output|Number} coin or output or value.
+   * @returns {Boolean} Whether the input was able to be signed.
+   */
+
+  signCustomInput(index, rings, redeemScript, witnessScript, type, coin) {
+    function getValue(tx, coin, input) {
```

Comment:
Since this function is meant to be called directly by a user / library application, I wonder if it should have more complete input validation. Especially (if I understand correctly) that it can only be used in SegWit transactions. For a single-key template you might also want to accept a single `ring` in addition to an array `[ring]`. Similar methods like `mtx.sign()` and `mtx.template()` do this.